src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -989,19 +989,18 @@ public override int GetHashCode()
 
             if (_bits == null)
                 return _sign;
-            int hash = _sign;
+
+            HashCode hash = default;
+            hash.Add(_sign);
             for (int iv = _bits.Length; --iv >= 0;)
-                hash = NumericsHelpers.CombineHash(hash, unchecked((int)_bits[iv]));
-            return hash;
+                hash.Add(_bits[iv]);
+            return hash.ToHashCode();
         }
 
         public override bool Equals([NotNullWhen(true)] object? obj)
         {
             AssertValid();
-
-            if (!(obj is BigInteger))
-                return false;
-            return Equals((BigInteger)obj);
+            return obj is BigInteger bigInt && Equals(bigInt);
```

Comment:
nit: This matches the parameter name and what we use in a few other places
```suggestion
            return obj is BigInteger other && Equals(other);
```